components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/internal/SecretManagerComponent.java

```diff
@@ -0,0 +1,142 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.secret.mgt.core.internal;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.osgi.framework.BundleContext;
+import org.osgi.service.component.ComponentContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ReferenceCardinality;
+import org.osgi.service.component.annotations.ReferencePolicy;
+import org.wso2.carbon.identity.application.authentication.framework.util.FrameworkUtils;
+import org.wso2.carbon.identity.secret.mgt.core.SecretManager;
+import org.wso2.carbon.identity.secret.mgt.core.SecretManagerImpl;
+import org.wso2.carbon.identity.secret.mgt.core.SecretResolveManager;
+import org.wso2.carbon.identity.secret.mgt.core.SecretResolveManagerImpl;
+import org.wso2.carbon.identity.secret.mgt.core.dao.SecretDAO;
+import org.wso2.carbon.identity.secret.mgt.core.dao.impl.CachedBackedSecretDAO;
+import org.wso2.carbon.identity.secret.mgt.core.dao.impl.SecretDAOImpl;
+import org.wso2.carbon.identity.secret.mgt.core.model.SecretManagerConfigurationHolder;
+import org.wso2.carbon.user.core.service.RealmService;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+
+/**
+ * OSGi declarative services component which handles registration and un-registration of configuration management
+ * service.
+ */
+@Component(
+        name = "carbon.secret.mgt.component",
+        immediate = true
+)
+public class SecretManagerComponent {
+
+    private static final Log log = LogFactory.getLog(SecretManagerComponent.class);
+    private List<SecretDAO> secretDAOS = new ArrayList<>();
+
+    /**
+     * Register SecretManager as an OSGI service.
+     *
+     * @param componentContext OSGI service component context.
+     */
+    @Activate
+    protected void activate(ComponentContext componentContext) {
+
+        try {
+            BundleContext bundleContext = componentContext.getBundleContext();
+
+            SecretDAO secretDAO = new SecretDAOImpl();
+            bundleContext.registerService(SecretDAO.class.getName(), secretDAO, null);
+            bundleContext.registerService(SecretDAO.class.getName(),
+                    new CachedBackedSecretDAO(secretDAO), null);
+            SecretManagerConfigurationHolder secretManagerConfigurationHolder =
+                    new SecretManagerConfigurationHolder();
+            secretManagerConfigurationHolder.setSecretDAOS(secretDAOS);
+
+            bundleContext.registerService(SecretManager.class.getName(),
+                    new SecretManagerImpl(secretManagerConfigurationHolder), null);
```

Comment:
No need to pass the `secretManagerConfigurationHolder`. We can always get this instance using getInstance.

Reply:
`SecretManagerConfigurationHolder` should be inside internal package and have getInstance method. This will make sure only one instance of SecretManagerConfigurationHolder is available in the runtime

Reply:
resolved